trappy/ftrace.py

```diff
@@ -76,6 +108,22 @@ def __init__(self, name="", normalize_time=True, scope="all",
             setattr(self, attr, trace_class)
             self.trace_classes.append(trace_class)
 
+        cache_path = _trace_cache_path(self.trace_path)
+        cache_good = False
+
+        if _check_trace_cache(self.trace_path):
+            cache_good = True
+            try:
+                # Read csv into frames
+                for trace_class in self.trace_classes:
+                    csv_file = cache_path + '/' + trace_class.unique_word[:-1] + '.csv'
+                    trace_class.read_csv(csv_file)
+            except:
+                cache_good = False
+
+        if cache_good:
+            return
```

Comment:
If I'm not mistaken you can get rid of `cache_good`:
```
if _check_trace_cache(self.trace_path):
    try:
        for trace_class in self.trace_classes:
            csv_file = cache_path + '/' + trace_class.unique_word[:-1] + '.csv'
            trace_class.read_csv(csv_file)
        return
    except:
        pass
````

Also what if we e.g. successfully parse the CSV for trace class A (so its `data_frame` member is populated), then fail while parsing the CSV for class B? It looks like all the events for A will be duplicated in `__parse_trace_file` below.